xbmc/guilib/GUILabelControl.cpp

```diff
@@ -203,7 +210,15 @@ bool CGUILabelControl::OnMessage(CGUIMessage& message)
     }
   }
   if (message.GetMessage() == GUI_MSG_REFRESH_TIMER && IsVisible())
+  {
     UpdateInfo();
+  }
+  else if (message.GetMessage() == GUI_MSG_WINDOW_RESIZE && IsVisible())
+  {
+    // If the gui font size/aspect has been rescaled (e.g. screen calibration
+    // window) we have to refresh the text layout sizes
+    m_refreshLayout = true;
```

Comment:
Is this needed? Isn't it possible to simply call updateinfo and force refresh the layout there? What is the reason for the controlling bool? Is updateinfocalled on a loop?

Reply:
already tried but not works, seem this is like a cat and mice
from what i understand seem that kodi call from CApplication:
as first CGUIWindowSettingsScreenCalibration::DoProcess()
as second CGUIWindowManager::FrameMove()

when GUI_MSG_WINDOW_RESIZE is called by CGUIWindowSettingsScreenCalibration::FrameMove
if m_refreshLayout is not stored when CGUILabelControl::UpdateInfo is called (from DoProcess() ) the layout refresh seem not happen correclty

i have tried also move GUI_MSG_WINDOW_RESIZE call from CGUIWindowSettingsScreenCalibration::FrameMove to CGUIWindowSettingsScreenCalibration::DoProcess but without success

i have done a little scheme of the calls sequences to try understand better:
```
(1st call) CApplication call -> CGUIWindowSettingsScreenCalibration::DoProcess -> CGUIWindow::DoProcess -> CGUIControl::DoProcess -> CGUIControlGroup::Process
then CGUIControlGroup::Process execute for each control:
1st call - UpdateVisibility -> CGUIControl::UpdateVisibility -> UpdateInfo -> CGUILabelControl::UpdateInfo  (bInvalidated is true then RefreshLayoutSize)
2nd call - DoProcess -> CGUILabelControl::Process ->then  CGUIControl::DoProcess  update m_renderRegion (render region) ?? then bInvalidated will be false
3rd call - CGUIControl::Process

(2nd call) CApplication call -> FrameMove -> CGUIWindowManager::FrameMove() -> CGUIWindowSettingsScreenCalibration::FrameMove()
```

I have no other ideas at the moment